app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class Api::V1::CommentsController < Api::V1::ApplicationController
+  def index
+    comments = Post.find(params[:post_id]).comments
+    render json: comments, status: :ok
+  end
+
+  def create
+    comment = Comment.new(comment_params, post_id:, user_id:)
+    if comment.save
+      render json: comment, status: :created
+    else
+      render json: { errors: comment.errors }, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
```

Comment:
- [ Required ] Great job implementing the `fetch user posts` and `fetch post comments`, however, I was not able to add a comment to a post through this API `/api/v1/users/1/posts/1/comments` this is because the body of the request does not permit the `author_id` to be sent as part of the request body. You can make this code work by refactoring your code as shown below.

```ruby
   def create
     comment = Comment.new(text: comment_params['text'], 
                           author_id: comment_params['author_id'], 
                           post_id:  params['post_id'])
    if comment.save
      render json: comment, status: :created
    else
      render json: { errors: comment.errors }, status: :unprocessable_entity
    end
  end

  private

  def comment_params
    params.require(:comment).permit(:text, :author_id)
  end
```
## Current error 
![Screenshot from 2023-02-11 01-41-03](https://user-images.githubusercontent.com/16392046/218214597-244bc55b-158b-43c0-943c-9ec220892166.png)